client/hotrod/src/test/java/org/infinispan/hotrod/util/FlowUtils.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.hotrod.util;
+
+import java.util.List;
+import java.util.concurrent.Flow;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.rxjava3.core.Flowable;
+import org.reactivestreams.FlowAdapters;
+
+public final class FlowUtils {
+
+   private FlowUtils() { }
+
+   public static <T> Flowable<T> toFlowable(Flow.Publisher<T> publisher) {
+      return Flowable.fromPublisher(FlowAdapters.toPublisher(publisher));
+   }
+
+   public static <T> List<T> collect(Flowable<T> flowable) {
+      return flowable.toList()
+            .timeout(30, TimeUnit.SECONDS)
+            .blockingGet();
+   }
+
+   /**
+    * Collect all values in the {@link Flow.Publisher} into a list.
+    * This is a blocking method.
+    *
+    * @param publisher: Source of the values.
+    * @return A list with the elements from {@link Flow.Publisher}.
+    * @param <T>: Type of the elements.
+    */
+   public static <T> List<T> collect(Flow.Publisher<T> publisher) {
```

Comment:
```suggestion
   public static <R, A> R blockingCollect(Flow.Publisher<T> publisher, Collector<? super T, A, R> collector) {
```
This way the caller can use any type easier. Also if you want you could add the same method for list  that just passes Collectors.toList() to this one.